### DISCUSSION TITLE
Assistance with specific "static" layout + Is it possible to use marks for swallowing?

### DISCUSSION THREAD

**Discussion by wauterboi** [i3/i3#4947] [Q&A]
Hello,

I was hoping that I'd be able to use i3's tree saving functionality to accomplish a very specific layout, but I am running into some roadblocks as a newbie.

I'd like to create a layout like this:

```
1. Horizontal Split
  a. "Main Window" (Tabbed Split, 2/3 parent width)
  b. Vertical Split (1/3 parent width)
    i.   Tabbed Split (1/3 parent height)
    ii.  Tabbed Split (1/3 parent height)
    iii. Tabbed Split (1/3 parent height)
```

There's two ways I'd envision adding stuff to this split:

1. Using marks - I wasn't able to find any documentation to suggest that marks are available as criteria for swallowing, though
2. Manually selecting the placeholder and adding windows

Is it possible to use marks to automatically place windows in specific containers? Are there any other suggestions the community might be able to provide for implementing this layout?

Thanks for reading.

**Comment by Airblader** [comment]
Hi,

For marks to work with swallowing, two things need to happen: they need 
to be considered in the match, and they need to be set when the 
swallowing takes place. I think the former should work, i.e. we do 
consider marks when swallowing; however, we swallow when a window is 
first mapped, which will happen before any marks have been applied. 
Without looking at the code I'm not sure we re-trigger swallowing when 
marks change, but the easy answer is "just try it out".

However, I'm not sure I understand how that would help: in order to use 
marks for swallowing, you'd first have to set the mark on a window, 
which in turn means you need some selection process anyway.


Ingo

**Reply by wauterboi** [reply]
I'd hate to ask the obvious, but is it as simple as writing `"mark": "^namegoeshere$"` in the criteria object?

**Reply by Airblader** [reply]
This is from memory, but I believe we use the same data structre as matches in things like `for_window`, so it'd be `con_mark`.
